### ISSUE TITLE
MirroredStrategy compared to OneDeviceStrategy slower 

### ISSUE THREAD

**Issue by olk** [tensorflow/tensorflow#35095]
**System information**
- OS Platform and Distribution: Arch Linux, 5.4.2-arch1-1-ARCH
- TensorFlow installed from: binary
- TensorFlow version: 2.1.0rc0-1
- Keras version: 2.2.4-tf
- Python version: 3.8
- GPU model and memory: 2x GTX 1080 Ti 11GB"`

**Describe the current behavior**
executing Tensorflow's MNIST handwriting example with MirroredStrategy using two GPUs is slower (286 sec) than OneDeviceStrategy (197 sec) 

**Describe the expected behavior**
using MirroredStrategy should be faster or at least fast as OneDeviceStrategy

**Cod e to reproduce the issue**
hype rparameters were tuned for MirroredStrategy using Keras Tuner

```
  import tensorflow as tf
  import tensorflow_datasets as tfds
  import time
  
  from tensorflow.keras.optimizers import Adam
  
  def build_model():
      filters = 48
      units = 24
      kernel_size = 7
      learning_rate = 1e-4
      model = tf.keras.Sequential([
        tf.keras.layers.Conv2D(filters=filters, kernel_size=(kernel_size, kernel_size), activation='relu', input_shape=(28, 28, 1)),
        tf.keras.layers.MaxPooling2D(),
        tf.keras.layers.Flatten(),
        tf.keras.layers.Dense(units, activation='relu'),
        tf.keras.layers.Dense(10, activation='softmax')
      ])
      model.compile(loss='sparse_categorical_crossentropy', optimizer=Adam(learning_rate), metrics=['accuracy'])
      return model
  
  datasets, info = tfds.load(name='mnist', with_info=True, as_supervised=True)
  mnist_train, mnist_test = datasets['train'], datasets['test']
  
  num_train_examples = info.splits['train'].num_examples
  num_test_examples = info.splits['test'].num_examples
  
  strategy = tf.distribute.OneDeviceStrategy(device='/gpu:0')
  #strategy = tf.distribute.MirroredStrategy()
  print('Number of devices: {}'.format(strategy.num_replicas_in_sync))
  
  BUFFER_SIZE = 10000
  BATCH_SIZE = 32
  
  def scale(image, label):
    image = tf.cast(image, tf.float32)
    image /= 255
    return image, label
  
  train_dataset = mnist_train.map(scale).shuffle(BUFFER_SIZE).repeat().batch(BATCH_SIZE).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)
  eval_dataset = mnist_test.map(scale).repeat().batch(BATCH_SIZE).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)
  
  with strategy.scope():
    model = build_model()
  
  epochs=5
  start = time.perf_counter()
  model.fit(
          train_dataset,
          validation_data=eval_dataset,
          steps_per_epoch=num_train_examples/epochs,
          validation_steps=num_test_examples/epochs,
          epochs=epochs)
  elapsed = time.perf_counter() - start
  print('elapsed: {:0.3f}'.format(elapsed))
```

**Other info / logs**
**OneDeviceStrategy**:
> ..
> 12000/12000 [==============================] - 39s 3ms/step - loss: 0.0089 - accuracy: 0.9978 - val_loss: 0.0385 - val_accuracy: 0.9892
> elapsed: 197.177

**MirroredStrategy**:
> ...
> 12000/12000 [==============================] - 56s 5ms/step - loss: 0.0098 - accuracy: 0.9977 - val_loss: 0.0390 - val_accuracy: 0.9884
> elapsed: 286.224

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.1 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/200f60befed7b2622b775b9fff6f90b0/untitled483.ipynb). Thanks!

**Comment by guptapriya** [comment]
@ravikyram how did you run with 2 GPUs in colab? virtual GPUs? That's not a good way to debug performance issues. 

@olk can you please provide profiles for the run with MirroredStrategy with your 2 GPUs? Also, what was the global batch size in both cases? To get similar performance, you will likely need to increase your per replica batch size. Moreover, for a small model like MNIST, it is not unexpected that overhead of synchronization will be high compared to step time.

**Comment by olk** [comment]
@guptapriya Sorry, but what do you mean with 'profiles for the run with MirroredStrategy' exactly?
The batch size was 32 for both cases - the batch size was part of the tuned hyper parameters (using Keras Tuner).

**Comment by neil-119** [comment]
I too am seeing this. My rig has two RTX 2080 Tis (invested in another one of these GPUs solely for ML work) so I would be very interested in a solution.

**Comment by olk** [comment]
@neil-119 I'm switching to MXNet - it does much better utilize GPUs (including multi GPU onfig) + less memory consumption at the GPUs. Beside [Gluon](https://medium.com/@julsimon/gluon-building-blocks-for-your-deep-learning-universe-4bce4e56ef55) - [Keras](https://github.com/awslabs/keras-apache-mxnet) works with MXNet too.

**Comment by ymodak** [comment]
Closing this issue now. Thanks!
